cmd/oras/internal/option/common.go

```diff
@@ -17,25 +17,44 @@ package option
 
 import (
 	"context"
+	"errors"
+	"os"
 
 	"github.com/sirupsen/logrus"
 	"github.com/spf13/pflag"
+	"golang.org/x/term"
 	"oras.land/oras/internal/trace"
 )
 
 // Common option struct.
 type Common struct {
 	Debug   bool
 	Verbose bool
+	TTY     *os.File
+
+	avoidTTY bool
 }
 
 // ApplyFlags applies flags to a command flag set.
 func (opts *Common) ApplyFlags(fs *pflag.FlagSet) {
 	fs.BoolVarP(&opts.Debug, "debug", "d", false, "debug mode")
 	fs.BoolVarP(&opts.Verbose, "verbose", "v", false, "verbose output")
+	fs.BoolVarP(&opts.avoidTTY, "no-tty", "", false, "[Preview] avoid using stdout as a terminal")
 }
 
 // WithContext returns a new FieldLogger and an associated Context derived from ctx.
 func (opts *Common) WithContext(ctx context.Context) (context.Context, logrus.FieldLogger) {
 	return trace.NewLogger(ctx, opts.Debug, opts.Verbose)
 }
+
+// Parse gets target options from user input.
+func (opts *Common) Parse() error {
+	f := os.Stderr
```

Comment:
Request documentation on setting `os.Stderr` as the output.

Reply:
What kind of docs are you suggesting? This is to accommodate the upcoming change for piped UX, so user can see the progress bar and use `-o json` at the same time.

Reply:
Added